### DISCUSSION TITLE
Calibration

### DISCUSSION THREAD

**Discussion by VesaOskari** [egzumer/uv-k5-firmware-custom#614] [Q&A] [answered]
I have read that each individual radio have its own factory written calibration values (tuning?). I have understood that this unique calibration data is between 0x1e00 and 0x2000. 

In this page it is written that Egzumer firmware does not overwrite this calibration data.
https://github.com/egzumer/uv-k5-firmware-custom/discussions/134

The size of Egzumer is about 60K. How is it possible that the firmware does not destroy this original calibration data and have effect on performance?

**Comment by prokrypt** [comment, marked as answer]
Firmware is stored on flash memory built into the microprocessor. Radio config (channels, etc) and calibration are stored on external eeprom.

**Reply by VesaOskari** [reply]
That makes sense. EEPROM and flash memory are two different things.

I have a linux PC and want to be sure I do a full backup before changing the firmware. K5prog will do that, I think. 

So to backup the configuration and calibration on eeprom I would use 
k5prog -r -v

Can I backup the current firmware like this?
k5prog -b -v

After making these backups I would use the flatpak version of Chromium to flash the new firmware. Or K5prog if that fails.

Edit: There is K5TOOL which seems to be more easy to use. This thread is useful: https://www.eevblog.com/forum/rf-microwave/quansheng-uv-k5(8)-firmware-modding/
